src/app/core/entity-details/form/form.component.ts

```diff
@@ -82,6 +105,25 @@ export class FormComponent<E extends Entity> implements FormConfig, OnInit {
     this.entityFormService.resetForm(this.form, this.entity);
     this.form.disable();
   }
+
+  private filterFieldGroupsByPermissions() {
+    this.filteredFieldGroups = this.fieldGroups
+      .map((group) => {
+        group.fields = group.fields.filter((field) =>
+          this.abilityService.hasReadPermission(this.entity, field),
+        );
+        return group;
+      })
+      .filter((group) => group.fields.length !== 0);
+  }
+
+  private disableReadOnlyFormControls() {
+    Object.keys(this.form.controls).forEach((fieldId) => {
+      if (!this.abilityService.hasEditPermission(this.entity, fieldId)) {
```

Comment:
```suggestion
      if (!this.ability.can('read', this.entity, fieldId)) {
```
This way you wouldn't event need the extra functions in the ability service.

Reply:
done